### DISCUSSION TITLE
GitHub Actions: Feedback to Starter workflows

### DISCUSSION THREAD

**Discussion by pulkitaggarwl** [community/community#8816] [Actions]
Feedback to Starter workflows

**Comment by CalvinRodo** [comment]
It would be nice if we could do this at the org level. 

Have some predefined workflows in a repo that are available to folks instead of having them copy and paste them.

**Reply by anuragc617** [reply]
@CalvinRodo you can add workflow at org level and those workflows can be used in the org's repositories. These org workflows are also shown in Starter workflows. Follow this [doc](https://docs.github.com/en/actions/learn-github-actions/creating-workflow-templates#creating-a-workflow-template) to add templates to your org.

**Reply by MelindaMalmgren** [reply]
Is this expected to work for enterprise orgs? I followed the instructions in your link, but my starter workflow is not suggested when creating a workflow within my org.

**Reply by anuragc617** [reply]
No, it works for all type of orgs and users account as well. There are few things to consider:
1. .github repository should be public. 
2. the workflow yaml name as well as properties.json file name should be same.
3. If iconName is present in the properties.json then SVG icon file should be present 

On starter workflows side:
1. In starter workflow, "Suggested for this repository" section will show your org workflows if repo language matches with workflow properties.
2. Otherwise, all workflows will be shown in "By <orgs name>" Section.

**Reply by scruplelesswizard** [reply]
For those with Github Enterprise Cloud it is unexpected that the `.github` repo must be Public. This significantly limits the user experience. Expected behavior is that an Org's `.github` Internal repo would behave the same.
Many orgs are unable or unwilling to expose their `.github` repo publicly

Also related https://github.com/community/community/discussions/8816#discussioncomment-2726243

**Comment by kokes** [comment]
I tried it on a few repos and it worked wonderfully. One tiny thing I think could help: (optional?) matrix strategies for both runtimes and operating systems. For the languages I tried (Go, Python) it defaulted to latest-language-release@ubuntu, but it'd be good if there was a way to run this against past three stable language releases on linux/windows/macos.

**Reply by ethomson** [reply]
This makes a lot of sense - this is something that the Node.js starter workflow does, it would be nice to see it applied more broadly.  Thanks for the feedback @kokes.

**Comment by ghost** [comment]
Hey,
I have an issue related to the github actions, When I open the actions tab of my repository; a server error always occur.
I've tried to edit the workflow but still no progress.

It opens the "Server Error . Github" tab in the browser instead of opening the action tab.

**Reply by ghost** [reply]
Sorry. Bad place to ask 😅

**Comment by bewuethr** [comment]
This is mentioned in [this thread](https://github.com/github/feedback/discussions/8816#discussioncomment-1838086), but I'd like to put it here as a top-level answer: it's currently not possible to use non-public starter workflows in an org, because they must be in the public `.github` repository, which is a bit of a bummer.

GitHub Enterprise allows reusing workflows between internal repositories, and recently, an internal-facing version of `.github` [has been introduced](https://github.blog/changelog/2022-04-20-organization-profile-updates-member-only-readmes-and-pinned-private-repositories/), `.github-private`. It'd be great if I could place private starter workflows in there (or, for an organization in an enterprise, have a third special repo, `.github-internal`, and store internally available starter workflows in there).

**Reply by badbort** [reply]
This is exactly what I'm after and I really hope this gets implemented.

I've set this up in a personal public org with success, but sadly when trying to do the same for my workplaces' org it doesn't seem possible and we definitely don't want to make the starter workflows public.

Hoping the github team gives us some better options for private starter worfklows for GitHub Enterprise users.

**Reply by gazab** [reply]
Agree. Until starter workflows can be made private this is unfortunately useless for us.

**Reply by dsdavis4** [reply]
I also agree with this. The docs are misleading and one would assume they could put starter workflows in `.github-private` repository, but this is not the case. Starter workflows are not useable for us till this is implemented

**Reply by roulettedares** [reply]
We really need the option to make the starter workflows repo internal/private.

**Comment by fty4** [comment]
It would be great to also have the option to create starter workflows on enterprise level.

On the GitHub Enterprise Server I am active on we have many organisations and it would be great to also share the workflow-templates within the whole enterprise.

Is it possible that this feature could be added at some point?
